src/python/pants/jvm/jdk_rules.py

```diff
@@ -216,6 +216,8 @@ async def prepare_jdk_environment(
     else:
         coursier_jdk_option = shlex.quote(f"--jvm={version}")
 
+    jvm_index = f"--jvm-index {coursier.jvm_index}"
```

Comment:
Does this work with the space instead of a `=`? The arguments are passed via an array of strings, thus I would think this should either be a single argument form like `f"--jvm-index={coursier.jvm_index}"` or should be two arguments `["--jvm-index", coursier.jvm_index]` and then use *jvm_index to insert it into the list later in the function.

Reply:
Also, would it be better to not pass this argument in the default case?

Reply:
It seems to be working as expected with the space.  The example usage for the argument in the Coursier documentation looks like this:
```
cs java --available --jvm-index https://url/of/your/index.json
```
So I chose to use a space.  I can check if it works with `=` as well if it would be better to have consistency with how the version argument is formatted in line 217.

I like the idea of not passing the argument by default.  In that case would it be better for the default value of the option to be an empty string?

Reply:
Ah, I think the space does work as is, but this is formatted into a shell script directly (and thus the space works because it is relying on shell expansion/argument parsing).

Thus, the space working is revealing a deeper problem: it should be created with `shlex.quote` so that it behaves properly if the URL contains any characters that are special to a shell (if not, special characters like `;` will result in at best incorrect results, and at worse shell-injection vulnerabilities).

Thus, I think it should either, look similar to the `coursier_jdk_option` assignment on the previous line:

```suggestion
    jvm_index = shlex.quote(f"--jvm-index={coursier.jvm_index}")
```

(NB. the `=` then becomes required, or else it's like calling `cs java --available '--jvm-index https://example.com'`, which doesn't seem to work for me.)

Or, with the two argument version:

```suggestion
    jvm_index = ["--jvm-index", shlex.quote(coursier.jvm_index)]
```

Reply:
I think the shell quoting ought to be applied once, later. So this issue would've been caught rather than sprinkling with quoting calls.

Reply:
> I think the shell quoting ought to be applied once, later (to all arguments). So this issue would've been caught rather than sprinkling with quoting calls.

I agree. Quoting with `shlex.quote` when we construct the final command will ensure less chance of problems in the future.

Reply:
Thank you for the suggestion.  I've looked into it a bit, but I'm not sure how to proceed with applying `shlex.quote` only once.  Quoting the entire command seems to cause errors when it's executed.

I was thinking I could apply `shlex.quote` to each value in `coursier_options` in a separate statement, or I could modify the prefixed function like this:
``` python
def prefixed(arg: str) -> str:
    if arg.startswith("__"):
        return f"${{PANTS_INTERNAL_ABSOLUTE_PREFIX}}{arg}"
    else:
        return shlex.quote(arg)
```
So at least any future additional options would be quoted when `optionally_prefixed_coursier_args` is defined.  Would that be sufficient?

Reply:
@kaos @tdyas Could either of you check my previous response?  I think I may have misunderstood your feedback, so I would like to confirm before making a change.

Reply:
How about quoting the args here:
https://github.com/pantsbuild/pants/blob/fe449a537423d6fa25ef6887e28094b1ec6959c5/src/python/pants/jvm/resolve/coursier_setup.py#L183-L189

```diff
    def args(self, args: Iterable[str], *, wrapper: Iterable[str] = ()) -> tuple[str, ...]:
        return (
            self.post_process_stderr,
            *wrapper,
            os.path.join(self.bin_dir, self.coursier.exe),
--          *args,
++          *map(shlex.quote, args),
        )
```

Reply:
Nice, I'll give that a try.  Thank you!